posydon/binary_evol/DT/step_detached.py

```diff
@@ -1503,10 +1508,13 @@ def get_omega(star, is_secondary = True):
                 print("solution of ODE", s)
             if s.status == -1:
                 print("Integration failed", s.message)
-                binary.state += ' (Integration failure)'
-                # binary.event = "END"
+
+                if binary.state in ['detached']:
+                    binary.state += ' (Integration failure)'
+                else:
+                    raise ValueError("need to add new binary state to flow chart for evolution to end correctly ",
+                                "when detached integration fails: ", binary.state, " (Integration failure)")
                 return
```

Comment:
Just a style point: why do you have the return here behind the raising of the error, while in line 1182 you put it only in the other case? I don't have a strong preference, but it should be consistent, hence may check which way is otherwise used in the code.

Reply:
Yeah, I agree it should be consistent, I've moved it inside the code block.